search/api.py

```diff
@@ -73,6 +81,48 @@ def gen_course_id(course_id):
     return "co_{}".format(safe_id)
 
 
+def gen_bootcamp_id(bootcamp_id):
+    """
+    Generates the Elasticsearch document id for a bootcamp
+
+    Args:
+        bootcamp_id (str): The course_id of a Bootcamp object
+
+    Returns:
+        str: The Elasticsearch document id for this object
+    """
+    safe_id = urlsafe_b64encode(bootcamp_id.encode("utf-8")).decode("utf-8").rstrip("=")
```

Comment:
Is there a reason for stripping the `=` here? I think you can leave it in so we can reverse it.

Reply:
I don't know the range of possible contents for a Bootcamp's `course_id`, so I don't know for sure. The code copies the logic of `gen_course_id`, as bootcamps are course-like. If you think there's a good reason we should keep it for `gen_course_id` but not for `gen_bootcamp_id`, we can remove the strip for `=` here.

Reply:
I didn't see that. In that case, keep it here and add it to `gen_program_id` and `gen_user_list_id` since those dropped the `rstrip()` call

Reply:
We discussed yesterday that it would be fine to use the autogenerated Django object id for Programs and UserLists, since they don't have a reliably generated unique id like `course_id` that we source. As such, the inputs will likely just be `ints`, and probably don't need the `rstrip()` calls. It's confusing, and I'm going to add comments. Do you think we should still put it in?

Reply:
yeah, if that's the case then the id can just go directly into the generated id and not be encoded.